mindsdb/integrations/handlers/surrealdb_handler/surrealdb_handler.py

```diff
@@ -0,0 +1,231 @@
+        except Exception as e:
+            log.logger.error(f"Error while connecting to SurrealDB, {e}")
+
+        return self.connection
+
+    def check_connection(self) -> StatusResponse:
+        """
+        Check connection to the handler.
+        Returns:
+            HandlerStatusResponse
+        """
+        response_code = StatusResponse(False)
+        need_to_close = self.is_connected is False
+        try:
+            self.connect()
+            response_code.success = True
+        except Exception as e:
+            log.logger.error(f'Error connecting to SurrealDB, {e}!')
+            response_code.error_message = str(e)
+        finally:
+            if response_code.success is True and need_to_close:
+                self.disconnect()
+            if response_code.success is False and self.is_connected is True:
+                self.is_connected = False
+
+        return response_code
+
+    def disconnect(self):
+        """
+        Close the existing connection to the SurrealDB database
+        """
+        if self.is_connected is False:
+            return
+        try:
+            self.connection.close()
+            self.is_connected = False
+        except Exception as e:
+            log.logger.error(f"Error while disconnecting to SurrealDB, {e}")
+
+        return
+
+    def native_query(self, query: str) -> Response:
+        """
+        Receive raw query and act upon it somehow.
+        Args:
+            query (Any): query in SurrealQL to execute
+        Returns:
+            HandlerResponse
+        """
+        need_to_close = self.is_connected is False
+        conn = self.connect()
+        cur = conn.cursor()
+        try:
+            cur.execute(query)
+            result = cur.fetchall()
+            if result:
+                response = Response(
+                    RESPONSE_TYPE.TABLE,
+                    data_frame=pd.DataFrame(
+                        result,
+                        columns=[x[0] for x in cur.description],
+                    )
+                )
+            else:
+                response = Response(RESPONSE_TYPE.OK)
+        except Exception as e:
+            log.logger.error(f'Error running query: {query} on SurrealDB!')
+            response = Response(
+                RESPONSE_TYPE.ERROR,
+                error_message=str(e)
+            )
+
+        cur.close()
+
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    def query(self, query: ASTNode) -> Response:
+        """
+        Receive query as AST (abstract syntax tree) and act upon it somehow.
+        Args:
+            query (ASTNode): sql query represented as AST. It may be any kind
+                of query: SELECT, INSERT, DELETE, etc
+        Returns:
+            HandlerResponse
+        """
+        query_string = query.to_string()
+
+        # ensure the correct query is passed
+        last_word = query_string.split()[-1]
```

Comment:
Thanks @manosdell. Can you please clarify why we need these two lines? Is there some existing surrealdb dialect that we can re-use for SQLAlchemy?

Reply:
Apart from this looks great and it is ready for merge

Reply:
Unfortunately such dialect doesn't seem to exist yet. SurrealDB is still in Beta after all. I added the second line because otherwise the select query wouldn't return only the specified columns from the database.

Reply:
Ok, let's start with what we have. We can improve it after this. Thanks great job